public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -597,7 +599,20 @@ function errorAndDurationQuery(
   let serviceGraphViewMetrics = [];
   let errorRateBySpanName = '';
   let durationsBySpanName: string[] = [];
-  const spanNames = getEscapedSpanNames(rateResponse.data[0][0]?.fields[1]?.values ?? []);
+
+  let labels = [];
+  if (request.app === CoreApp.Explore) {
+    if (rateResponse.data[0][0]?.fields[1]?.values) {
+      labels = rateResponse.data[0][0]?.fields[1]?.values;
+    }
+  } else if (rateResponse.data[0]) {
+    rateResponse.data[0].map((df: DataFrame) => {
```

Comment:
If you are not returning anything from `map` I think it would be better to use `forEach` to be clear it has a side effect.